libraries/AC_WPNav/AC_WPNav.cpp

```diff
@@ -992,6 +1018,10 @@ bool AC_WPNav::set_spline_origin_and_destination(const Vector3f& origin, const V
     _flags.segment_type = SEGMENT_SPLINE;
     _flags.new_wp_destination = true;   // flag new waypoint so we can freeze the pos controller's feed forward and smooth the transition
 
+    // initialise yaw related variables
+    _track_length_xy = safe_sqrt(sq(_destination.x - _origin.x)+sq(_destination.y - _origin.y));  // horizontal track length (used to decide if we should update yaw)
+    _flags.wp_yaw_set = false;
```

Comment:
Is this last line needed here? Question comes from the fact it is done here and in `wp_and_spline_init` too but not in `set_wp_origin_and_destination`. From looking at code I'd say it needs to be done in both `set_X_origin_and_destination` methods. Should it be done in `advance_X_target_along_track` methods too when none of the conditions is met?